pom.xml

```diff
@@ -1013,35 +1061,84 @@
 									</goals>
 									<configuration>
 										<backend>html5</backend>
+										<sourceDocumentExtensions>
+											<sourceDocumentExtension>.adoc</sourceDocumentExtension>
+											<sourceDocumentExtension>.htmladoc</sourceDocumentExtension>
+										</sourceDocumentExtensions>
+										<outputDirectory>${project.build.directory}/generated-docs/reference/html
+										</outputDirectory>
 										<sourceHighlighter>highlight.js</sourceHighlighter>
 										<doctype>book</doctype>
 										<attributes>
-											// these attributes are required to use the doc resources
-											<docinfo>shared</docinfo>
-											<stylesdir>css/</stylesdir>
-											<stylesheet>spring.css</stylesheet>
+											<highlightjsdir>js/highlight</highlightjsdir>
+											<highlightjs-theme>atom-one-dark-reasonable</highlightjs-theme>
 											<linkcss>true</linkcss>
+											<imagesdir>./images</imagesdir>
 											<icons>font</icons>
+											<stylesdir>css/</stylesdir>
+											<stylesheet>spring.css</stylesheet>
+											<attribute-missing>warn</attribute-missing>
+										</attributes>
+										<logHandler>
+											<outputToConsole>true</outputToConsole>
+											<failIf>
+												<!--												<severity>DEBUG</severity>-->
+											</failIf>
+										</logHandler>
+									</configuration>
+								</execution>
+								<execution>
+									<id>generate-htmlsingle-documentation</id>
+									<phase>prepare-package</phase>
+									<goals>
+										<goal>process-asciidoc</goal>
+									</goals>
+									<configuration>
+										<backend>html5</backend>
+										<sourceDocumentExtensions>
+											<sourceDocumentExtension>.htmlsingleadoc</sourceDocumentExtension>
+										</sourceDocumentExtensions>
+										<outputDirectory>
+											${project.build.directory}/generated-docs/reference/htmlsingle
+										</outputDirectory>
+										<sourceHighlighter>highlight.js</sourceHighlighter>
+										<doctype>book</doctype>
+										<attributes>
 											<highlightjsdir>js/highlight</highlightjsdir>
 											<highlightjs-theme>atom-one-dark-reasonable</highlightjs-theme>
-											<allow-uri-read>true</allow-uri-read>
-											<nofooter/>
-											<toc>left</toc>
```

Comment:
We need toc: left and toc-levels: 4. They are part of how the dynamic ToC is configured. (4 may be the default for toc-levels, though.)